[Futures] Bump log verbosity when ignoring cb errors in python future.

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#54476 [Futures] Bump log verbosity when ignoring cb errors in python future.**
* #54475 [Futures] set completed_ after postMarkCompletedHook

Per title. For `add_done_callback`, we log but swallow exceptions in order to keep consistent with what concurrent.futures python library does, see discussion in https://github.com/pytorch/pytorch/pull/45675.

Although, it would be good to improve the verbosity here as this can be a source of confusion if users are setting a different future via `add_done_callback`, and an error is hit resulting in an unexpected hang (see https://github.com/pytorch/pytorch/issues/52132 for more details on how this can happen).

Differential Revision: [D27253004](https://our.internmc.facebook.com/intern/diff/D27253004/)

rohan-varma committed:
[Futures] Bump log verbosity when ignoring cb errors in python future.

Per title. For `add_done_callback`, we log but swallow exceptions in order to keep consistent with what concurrent.futures python library does, see discussion in https://github.com/pytorch/pytorch/pull/45675.

Although, it would be good to improve the verbosity here as this can be a source of confusion if users are setting a different future via `add_done_callback`, and an error is hit resulting in an unexpected hang (see https://github.com/pytorch/pytorch/issues/52132 for more details on how this can happen).

Differential Revision: [D27253004](https://our.internmc.facebook.com/intern/diff/D27253004/)

[ghstack-poisoned]

lw reviewed (approved):
Nice! This makes total sense to me!

I'd even say we could go as high up as a warning? An uncaught exception is, well, unexpected, right? And if the user doesn't like getting those warnings in their logs they could catch these exceptions on their own (or not raise them)?

As a side question: is there actually a way to get those `VLOG(1)` messages printed at runtime? We have a ton of them in the TensorPipe agent but I tried a whole lot of combinations of glog environment variables and cmdline flags but couldn't get them to print.

rohan-varma commented:
@lw Yeah, I haven't been able to figure out how to get VLOGs to show up as well, though admittedly I haven't tried very hard and usually switch them to a LOG(INFO) when debugging locally. It was also a pain point in debugging issues for a few other RPC issues, so switching them out seems to be the easiest way forward.

rohan-varma committed:
Update on "[Futures] Bump log verbosity when ignoring cb errors in python future."


Per title. For `add_done_callback`, we log but swallow exceptions in order to keep consistent with what concurrent.futures python library does, see discussion in https://github.com/pytorch/pytorch/pull/45675.

Although, it would be good to improve the verbosity here as this can be a source of confusion if users are setting a different future via `add_done_callback`, and an error is hit resulting in an unexpected hang (see https://github.com/pytorch/pytorch/issues/52132 for more details on how this can happen).

Differential Revision: [D27253004](https://our.internmc.facebook.com/intern/diff/D27253004/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[Futures] Bump log verbosity when ignoring cb errors in python future."


Per title. For `add_done_callback`, we log but swallow exceptions in order to keep consistent with what concurrent.futures python library does, see discussion in https://github.com/pytorch/pytorch/pull/45675.

Although, it would be good to improve the verbosity here as this can be a source of confusion if users are setting a different future via `add_done_callback`, and an error is hit resulting in an unexpected hang (see https://github.com/pytorch/pytorch/issues/52132 for more details on how this can happen).

Differential Revision: [D27253004](https://our.internmc.facebook.com/intern/diff/D27253004/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[Futures] Bump log verbosity when ignoring cb errors in python future."


Per title. For `add_done_callback`, we log but swallow exceptions in order to keep consistent with what concurrent.futures python library does, see discussion in https://github.com/pytorch/pytorch/pull/45675.

Although, it would be good to improve the verbosity here as this can be a source of confusion if users are setting a different future via `add_done_callback`, and an error is hit resulting in an unexpected hang (see https://github.com/pytorch/pytorch/issues/52132 for more details on how this can happen).

Differential Revision: [D27253004](https://our.internmc.facebook.com/intern/diff/D27253004/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[Futures] Bump log verbosity when ignoring cb errors in python future."


Per title. For `add_done_callback`, we log but swallow exceptions in order to keep consistent with what concurrent.futures python library does, see discussion in https://github.com/pytorch/pytorch/pull/45675.

Although, it would be good to improve the verbosity here as this can be a source of confusion if users are setting a different future via `add_done_callback`, and an error is hit resulting in an unexpected hang (see https://github.com/pytorch/pytorch/issues/52132 for more details on how this can happen).

Differential Revision: [D27253004](https://our.internmc.facebook.com/intern/diff/D27253004/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[Futures] Bump log verbosity when ignoring cb errors in python future."


Per title. For `add_done_callback`, we log but swallow exceptions in order to keep consistent with what concurrent.futures python library does, see discussion in https://github.com/pytorch/pytorch/pull/45675.

Although, it would be good to improve the verbosity here as this can be a source of confusion if users are setting a different future via `add_done_callback`, and an error is hit resulting in an unexpected hang (see https://github.com/pytorch/pytorch/issues/52132 for more details on how this can happen).

Differential Revision: [D27253004](https://our.internmc.facebook.com/intern/diff/D27253004/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[Futures] Bump log verbosity when ignoring cb errors in python future."


Per title. For `add_done_callback`, we log but swallow exceptions in order to keep consistent with what concurrent.futures python library does, see discussion in https://github.com/pytorch/pytorch/pull/45675.

Although, it would be good to improve the verbosity here as this can be a source of confusion if users are setting a different future via `add_done_callback`, and an error is hit resulting in an unexpected hang (see https://github.com/pytorch/pytorch/issues/52132 for more details on how this can happen).

Differential Revision: [D27253004](https://our.internmc.facebook.com/intern/diff/D27253004/)

[ghstack-poisoned]

rohan-varma commented:
build error is unrelated - 

```
error: failed to execute compile
caused by: error reading compile response from server
caused by: Failed to read response header
caused by: failed to fill whole buffer
[1276/4167] Building C object confu-deps\XNNPACK\CMakeFiles\XNNPACK.dir\src\f32-vbinary\gen\vsqrdiffc-avx-x16.c.obj
FAILED: confu-deps/XNNPACK/CMakeFiles/XNNPACK.dir/src/f32-vbinary/gen/vsqrdiffc-avx-x16.c.obj 
C:\w\b\windows\tmp_bin\sccache-cl.exe  /nologo -DCPUINFO_SUPPORTED_PLATFORM=1 -DFXDIV_USE_INLINE_ASSEMBLY=0 -DTH_BLAS_MKL -DWIN32_LEAN_AND_MEAN -DXNN_ENABLE_ASSEMBLY=1 -DXNN_ENABLE_MEMOPT=1 -DXNN_ENABLE_SPARSE=1 -DXNN_LOG_LEVEL=5 -D_OPENMP_NOFORCE_MANIFEST -D_WIN32_WINNT=0x0601 -Drestrict="" -I..\..\third_party\XNNPACK\include -I..\..\third_party\XNNPACK\src -I..\..\third_party\cpuinfo\deps\clog\include -I..\..\third_party\cpuinfo\include -I..\..\third_party\pthreadpool\include -I..\..\third_party\FXdiv\include -I..\..\third_party\FP16\include -I..\..\third_party\protobuf\src -IC:\w\b\windows\mkl\include /DWIN32 /D_WINDOWS /w /bigobj /MDd /Z7 /Ob0 /Od /RTC1 /w /bigobj /wd4146  /arch:AVX /showIncludes /Foconfu-deps\XNNPACK\CMakeFiles\XNNPACK.dir\src\f32-vbinary\gen\vsqrdiffc-avx-x16.c.obj /Fdconfu-deps\XNNPACK\CMakeFiles\XNNPACK.dir\XNNPACK.pdb /FS -c ..\..\third_party\XNNPACK\src\f32-vbinary\gen\vsqrdiffc-avx-x16.c
error: failed to execute compile
caused by: error reading compile response from server
caused by: Failed to read response header
caused by: failed to fill whole buffer
```

rohan-varma committed:
Update on "[Futures] Bump log verbosity when ignoring cb errors in python future."


Per title. For `add_done_callback`, we log but swallow exceptions in order to keep consistent with what concurrent.futures python library does, see discussion in https://github.com/pytorch/pytorch/pull/45675.

Although, it would be good to improve the verbosity here as this can be a source of confusion if users are setting a different future via `add_done_callback`, and an error is hit resulting in an unexpected hang (see https://github.com/pytorch/pytorch/issues/52132 for more details on how this can happen).

Differential Revision: [D27253004](https://our.internmc.facebook.com/intern/diff/D27253004/)

[ghstack-poisoned]
